site/src/components/FeatureStageBadge/FeatureStageBadge.tsx

```diff
@@ -0,0 +1,183 @@
+import type { Interpolation, Theme } from "@emotion/react";
+import Link from "@mui/material/Link";
+import { visuallyHidden } from "@mui/utils";
+import { HelpTooltipContent } from "components/HelpTooltip/HelpTooltip";
+import { Popover, PopoverTrigger } from "components/Popover/Popover";
+import {
+	type FC,
+	type HTMLAttributes,
+	type ReactNode,
+	useEffect,
+	useState,
+} from "react";
+import { docs } from "utils/docs";
+
+/**
+ * All types of feature that we are currently supporting. Defined as record to
+ * ensure that we can't accidentally make typos when writing the badge text.
+ */
+const featureStageBadgeTypes = {
+	beta: "beta",
+	experimental: "experimental",
+} as const satisfies Record<string, ReactNode>;
+
+const styles = {
+	badge: (theme) => ({
+		// Base type is based on a span so that the element can be placed inside
+		// more types of HTML elements without creating invalid markdown, but we
+		// still want the default display behavior to be div-like
+		display: "block",
+		maxWidth: "fit-content",
+
+		// Base style assumes that small badges will be the default
+		fontSize: "0.75rem",
+
+		cursor: "default",
+		flexShrink: 0,
+		padding: "4px 8px",
+		lineHeight: 1,
+		whiteSpace: "nowrap",
+		border: `1px solid ${theme.roles.preview.outline}`,
+		color: theme.roles.preview.text,
+		backgroundColor: theme.roles.preview.background,
+		borderRadius: "6px",
+		transition:
+			"color 0.2s ease-in-out, border-color 0.2s ease-in-out, background-color 0.2s ease-in-out",
+	}),
+
+	badgeHover: (theme) => ({
+		color: theme.roles.preview.hover.text,
+		borderColor: theme.roles.preview.hover.outline,
+		backgroundColor: theme.roles.preview.hover.background,
+	}),
+
+	badgeLargeText: {
+		fontSize: "1rem",
+	},
+
+	badgeExtraSmallText: {
+		// Have to beef up font weight so that the letters still maintain the
+		// same relative thickness as all our other main UI text
+		fontWeight: 500,
+		fontSize: "0.625rem",
+	},
+
+	tooltipTitle: (theme) => ({
+		color: theme.palette.text.primary,
+		fontWeight: 600,
+		fontFamily: "inherit",
+		fontSize: 18,
+		margin: 0,
+		lineHeight: 1,
+		paddingBottom: "8px",
+	}),
+
+	tooltipDescription: {
+		margin: 0,
+		lineHeight: 1.4,
+		paddingBottom: "8px",
+	},
+
+	tooltipLink: {
+		fontWeight: 600,
+		lineHeight: 1.2,
+	},
+} as const satisfies Record<string, Interpolation<Theme>>;
+
+type FeatureStageBadgeProps = Readonly<
+	Omit<HTMLAttributes<HTMLSpanElement>, "children"> & {
+		type: keyof typeof featureStageBadgeTypes;
+		size?: "xs" | "sm" | "lg";
+		variant: "interactive" | "static";
```

Comment:
The original request was to make a badge that could be put in headers and dropdown menus. We changed our mind on the dropdown behavior, but I figured it wouldn't hurt to keep it

The static variant is for dropdowns to make sure you don't have dropdowns trigger inside of dropdowns

Reply:
I think `variant` is a bit vague and opaque for what we're actually after here: "does it have a tooltip or not"

we could just have a `disableTooltip` prop, for example, or maybe `disableInteraction` to be a bit more general

Reply:
Yeah, totally fair. For context, there were originally three variants, but then it eventually became two. A boolean does make sense now